packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -298,25 +272,65 @@ const EDiscovery = WebexPlugin.extend({
       resource: `reports/${reportId}/contents/summary/${spaceId}`,
       timeout: options.timeoutMs
     })
-      .then((res) => {
-        if (!this.contentSummaryBySpaceIdCache.has(reportId)) {
-          this.contentSummaryBySpaceIdCache.set(reportId, new Map());
-        }
+      .then((res) => this._writeSpacesToContentSummaryBySpaceIdCache(reportId, [res.body])
+        .then(() => Promise.resolve(res)));
+  },
+
+  /**
+   * The results of a getContentSummary or getContentSummaryBySpaceId request are written to the contentSummaryBySpaceIdCache
+   * since information for a space is likely to be accessed very frequently when decrypting activities
+   * @param {UUID} reportId - Id of the report which contains the relevant content summary
+   * @param {Array<Object>} spaces - List of the space objects to be written to the cache
+   * @returns {Promise} - Promise resolution indicating operation is complete
+   */
+  _writeSpacesToContentSummaryBySpaceIdCache(reportId, spaces) {
+    if (!reportId || !spaces || !spaces.length) {
+      return Promise.resolve();
+    }
+
+    if (!this.contentSummaryBySpaceIdCache.has(reportId)) {
+      // wipe the cache - we will only need to store one reports data at any one time
+      this.contentSummaryBySpaceIdCache.clear();
```

Comment:
I've added a basic expiry mechanism, since each downloader client can only retrieve content for one report at a time, we can clear the cache whenever a new report comes in